### ISSUE TITLE
Error 1619

### ISSUE THREAD

**Issue by MikeDFW** [microsoft/TeamsMsiOverride#4]
I'm having an issue with the path being null. I can confirm the client has Powershell 5.1

```
AllowMsiOverride policy is set to True
Current Teams Machine-Wide Installer version is 1.3.00.4461
Target Teams Machine-Wide Installer version is 1.4.00.26376
Using 32-bit MSI
This will be an override update
Join-Path : Cannot convert 'System.Object[]' to the type 'System.String' required by parameter 'ChildPath'. Specified
Copying .\\1.4.00.26376\Teams_windows.msi to ...method is not supported.
At C:\Windows\ccmcache\14n\CheckMsiOverride.ps1:514 char:30

+ $toMsi = Join-Path $TempPath $msiName
+                              ~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Join-Path], ParameterBinding
Exception
    + FullyQualifiedErrorId : CannotConvertArgument,Microsoft.PowerShell.Command
s.JoinPathCommand

New-Item : Cannot bind argument to parameter 'Path' because it is null.
At C:\Windows\ccmcache\14n\CheckMsiOverride.ps1:516 char:31
+ New-Item -ItemType File -Path $toMsi -Force | Out-Null
+                               ~~~~~~
    + CategoryInfo          : InvalidData: (:) [New-Item], ParameterBindingValid
ationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Mic
rosoft.PowerShell.Commands.NewItemComm
   and

About to perform deployment using this msiexec command:
 msiexec.exe /fav "" /quiet ALLUSERS=1 /l*v C:\WINDOWS\TEMP\msiOverrideCheck_msi
exec.log
ERROR: MsiExec failed with exit code 1619

```

Here's what the TeamsMsiOverrideTrace.txt shows:

```
Start Trace 10/20/2021 14:32:23
Script Version 1.0.2108.0604
Parameters AllowInstallOvertopExisting: False, OverwritePolicyKey: False
Environment IsSystemAccount: True, IsOS64Bit: True
Removing temp folder C:\WINDOWS\TEMP\TeamsMsiOverrideCheck
The policy key AllowMsiOverride is not set, setting HKLM:\Software\Policies\Microsoft\Office\16.0\Teams\AllowMsiOverride to 1...
AllowMsiOverride policy is set to True
Current Teams Machine-Wide Installer version is 1.3.00.4461
Target Teams Machine-Wide Installer version is 1.4.00.26376
UninstallReg1: {39AF0813-FA7B-4860-ADBE-93B9B214B914} {731F6BAA-A986-45A4-8936-7C3AAAAA760B}
UninstallReg2: {39AF0813-FA7B-4860-ADBE-93B9B214B914} {731F6BAA-A986-45A4-8936-7C3AAAAA760B}
ProductsRegLM: 3180FA93B7AF0684DAEB399B2B419B41 AAB6F137689A4A549863C7A3AAAA67B0
ProductsRegCU: 
PackageName: Teams.msi Teams_windows_x64.msi
Using 32-bit MSI
This will be an override update
Copying .\\1.4.00.26376\Teams_windows.msi to ...
About to perform deployment using this msiexec command:
 msiexec.exe /fav "" /quiet ALLUSERS=1 /l*v C:\WINDOWS\TEMP\msiOverrideCheck_msiexec.log
ERROR: MsiExec failed with exit code 1619
Removing temp folder C:\WINDOWS\TEMP\TeamsMsiOverrideCheck
End Trace 10/20/2021 14:32:25
```

**Comment by chtartMSFT** [comment]
It appears you have both the 32-bit and 64-bit MSI's installed, which the script is not handling well.  I can look to update the script to at least warn that this is the case and exit gracefully.
I'll have to consider if there is a good way to allow the script to work in this case.  It really should be cleaned up, however, or it could lead to further complications in the future I expect.  I think in cases like this you end up needing to uninstall twice and then reinstall the correct bitness, but that in turn negates the need for the script.

**Comment by MikeDFW** [comment]
Thanks for the fast response! How difficult would it be to add in a section to remove the 32bit version and then only upgrade the 64bit version?

**Comment by chtartMSFT** [comment]
Uninstallation opens a bit of a can of worms.  When you perform the MSI uninstall it will also uninstall the current user profile's Team instance.  It looks like you are running from SYSTEM so it avoids the problem, but if this were run the script within a user profile, their Teams client would terminate and Teams would be removed for them and it wouldn't reappear until the signed out of Windows and back in again.
I'll investigate what it would look like to add such an option, and could just add the caveats noted above in the docs.

**Comment by chtartMSFT** [comment]
[CheckMsiOverride.txt](https://github.com/microsoft/TeamsMsiOverride/files/7392555/CheckMsiOverride.txt)
@MikeDFW Could you test this version of the script using -Uninstall32Bit switch and see how it works for you? It was lightly tested so please use in a limited manner.

**Comment by MikeDFW** [comment]
Thank you for the file, and the help with this. I'll try to get this tested by tomorrow afternoon.

**Comment by MikeDFW** [comment]
The test using -Uninstall32Bit worked. Thank you!

```
Start Trace 10/22/2021 10:19:38
Script Version 1.0.2110.2101
Parameters Type: Share, PreviewRing: False, BaseShare: .\, OverrideVersion: , MsiFileName: , AllowInstallOvertopExisting: False, OverwritePolicyKey: False, FixRunKey: False
Environment IsSystemAccount: True, IsOS64Bit: True
Removing temp folder C:\WINDOWS\TEMP\TeamsMsiOverrideCheck
AllowMsiOverride policy is set to True
ProductsRegLM: 3180FA93B7AF0684DAEB399B2B419B41 AAB6F137689A4A549863C7A3AAAA67B0
ProductsRegCU: 
About to uninstall 32-bit MSI using this msiexec command:
 msiexec.exe /X {39AF0813-FA7B-4860-ADBE-93B9B214B914} /quiet /l*v C:\WINDOWS\TEMP\msiOverrideCheck_msiexecUninstall.log
MsiExec completed successfully.
Teams Machine-Wide Installer was not found.
UninstallReg1: {731F6BAA-A986-45A4-8936-7C3AAAAA760B}
UninstallReg2: {731F6BAA-A986-45A4-8936-7C3AAAAA760B}
ProductsRegLM: AAB6F137689A4A549863C7A3AAAA67B0
ProductsRegCU: 
PackageName: Teams_windows_x64.msi
Using the BaseShare check for an update and aquire the new MSI...
Using 64-bit MSI from BaseShare
Target Teams Machine-Wide Installer version is 1.4.00.26376
This will be an override update
Copying .\\1.4.00.26376\Teams_windows_x64.msi to C:\WINDOWS\TEMP\TeamsMsiOverrideCheck\Teams_windows_x64.msi...
About to perform deployment using this msiexec command:
 msiexec.exe /fav "C:\WINDOWS\TEMP\TeamsMsiOverrideCheck\Teams_windows_x64.msi" /quiet ALLUSERS=1 /l*v C:\WINDOWS\TEMP\msiOverrideCheck_msiexec.log
MsiExec completed successfully.
New Teams Machine Installer version is 1.4.00.26376
Deployment successful, installer is now at target version!
Removing temp folder C:\WINDOWS\TEMP\TeamsMsiOverrideCheck
End Trace 10/22/2021 10:19:50

```

**Comment by chtartMSFT** [comment]
I pushed this version live now.  It is available on version 1.0.2110.2101.
